src/app/shared/services/user-session.service.ts

```diff
@@ -29,18 +30,18 @@ export class UserSessionService implements OnDestroy {
 
   constructor(
     private authService: AuthService,
-    private http: CurrentUserHttpService,
+    private currentUserService: CurrentUserHttpService,
   ) {
-    this.subscription = this.authService.status$.pipe(
-      switchMap(auth => {
-        if (!auth.authenticated) return of<UserProfile|undefined>(undefined);
-        return this.http.getProfileInfo().pipe(
-          catchError(_e => EMPTY)
-        );
-      }),
-      distinctUntilChanged((p1, p2) => p1 === p2 || (!!p1 && !!p2 && p1.groupId === p2.groupId))
+    this.subscription = merge(
+      this.authService.status$.pipe(filter(auth => auth.authenticated)),
+      this.refresh$,
+    ).pipe(
+      switchMap(() => this.currentUserService.getProfileInfo().pipe(
+        catchError(_e => EMPTY)
+      )),
+      distinctUntilChanged((p1, p2) => p1 === p2 || (p1.groupId === p2.groupId))
     ).subscribe(profile => {
-      this.session$.next(profile ? { user: profile } : undefined);
+      this.session$.next({ user: profile });
```

Comment:
You completely change the behavior of session$ and so of the full app with that change as you remove the undefined from sessions...
It might be valid... or not... but you haven't changed all other parts of the app which depend on that so that's not correct.

Reply:
Right, and it probably breaks the logout behavior mostly

Reply:
Fixed

Reply:
Actually it worked, and it didn't change the behavior of session. `session$` relies on `authService.status$`, `session$` is undefined when `authService.status$` is `notAuthenticated()` and it can be undefined at 2 moments:
1. At start, because `auth.status$` is a Behavior subject with default value `notAuthenticated()`
2. At logout, but it won't change anything because a redirection is performed, thus leaving the app, and redirecting to the same app causing a re-initialisation, which creates a temporary user.

Beyond that, `auth.status$` will be of type `CookieAuthenticated | TokenAuthenticated`, which both have `authenticated: true`

One could argue that the the `session$` is not supposed to know that at logout a redirection is performed, but then the problem will be with the `userSession.user$` since it filters out `userSession.session$` undefined results. Currently, performing a logout will not update the `userSession.user$` subject. Plus, if the idea _was_ stay on the app at logout, a temporary user would need to be created anyway.

To me, since a temporary user is created when no existing user is logged, the user session should always be populated _except_ at app initialization, which is (not explicitly) the case currently.